dask/dataframe/core.py

```diff
@@ -762,6 +762,17 @@ def assign(self, **kwargs):
 
         return elemwise(_assign, self, *pairs, columns=list(df2.columns))
 
+    @wraps(pd.DataFrame.rename)
+    def rename(self, **kwargs):
+        if kwargs.get('index') is not None:
+            raise ValueError("Cannot rename index.")
+        columns = kwargs.get('columns')
+        df2 = pd.DataFrame(columns=self.column_info).rename(columns)
+        self.column_info = df2.columns
```

Comment:
Probably don't want to mutate here.  Instead `df2.columns` should probably be used as an input to `map_partitions`